libc/stdio_scanf.c

```diff
@@ -0,0 +1,4210 @@
+	TEST_ASSERT_EQUAL_CHAR('b', res[BUFF_LEN - 5]);
+	free(res);
+}
+
+
+TEST(stdio_scanf_rest, m_brackets)
+{
+/* Disabled because of issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/667 */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+
+	const char *buff = "loremIPSUM IPSUMdolor dolorSitAmet";
+	const char *buff2 = "123loremIPSUM IPSUMdolor123 dolor123SitAmet";
+	char *res1, *res2, *res3;
+
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, fscanf(filep, "%m[A-z] %m[A-Z] %m[a-z]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM", res1);
+	TEST_ASSERT_EQUAL_STRING("IPSUM", res2);
+	TEST_ASSERT_EQUAL_STRING("dolor", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vfscanfWrapper(filep, "%m[A-z] %m[A-Z] %m[a-z]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM", res1);
+	TEST_ASSERT_EQUAL_STRING("IPSUM", res2);
+	TEST_ASSERT_EQUAL_STRING("dolor", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vsscanfWrapper(buff, "%m[A-z] %m[A-Z] %m[a-z]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM", res1);
+	TEST_ASSERT_EQUAL_STRING("IPSUM", res2);
+	TEST_ASSERT_EQUAL_STRING("dolor", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(buff, "%m[A-z] %m[A-Z] %m[a-z]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM", res1);
+	TEST_ASSERT_EQUAL_STRING("IPSUM", res2);
+	TEST_ASSERT_EQUAL_STRING("dolor", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+
+	fclose(filep);
+	filep = fopen(TESTFILE_PATH, "w+");
+
+	fprintf(filep, "%s", buff2);
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, fscanf(filep, "%m[1-9] %m[^1-9] %m[1-9]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vfscanfWrapper(filep, "%m[1-9] %m[^1-9] %m[1-9]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vsscanfWrapper(buff2, "%m[1-9] %m[^1-9] %m[1-9]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(buff2, "%m[1-9] %m[^1-9] %m[1-9]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+}
+
+
+/* Tests with %mc are disabled because of problems with host-generic-pc on git-hub ci */
+IGNORE_TEST(stdio_scanf_rest, m_c)
+{
```

Comment:
```suggestion
/* Address sanitizer used on Ubuntu 22.04 fails in such case, because of the following issue:
 * https://github.com/llvm/llvm-project/issues/61768
 */
#ifdef __linux__
__attribute__((no_sanitize_address))
#endif
TEST(stdio_scanf_rest, m_c)
{
/* Disabled because of issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/667 */
#ifdef __phoenix__
	TEST_IGNORE();
#endif
```